Source/Core/Core/PowerPC/TableGen.cpp

```diff
@@ -0,0 +1,470 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <bitset>
+#include <cstdlib>
+#include <deque>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <string>
+#include <utility>
+#include <variant>
+#include <vector>
+
+struct Instruction
+{
+  std::string opname;
+  std::deque<std::string> dispatch;
+  Instruction(std::string opname_, std::deque<std::string> dispatch_)
+      : opname(std::move(opname_)), dispatch(std::move(dispatch_))
+  {
+  }
+};
+
+struct SubTable
+{
+  std::string name;
+};
+
+struct StartTableMarker
+{
+  std::string name;
+  int shift;
+  int length;
+  std::string description;
+};
+struct Empty
+{
+};
+
+using InputLine = std::variant<Instruction, SubTable, StartTableMarker, Empty>;
+
+constexpr char SEPARATOR = ';';
+
+enum
+{
+  OPID_RANGES = 1,
+  DECODING_TABLE = 2,
+  OPNAME_TABLE = 4,
+};
+
+struct ColumnOptions
+{
+  unsigned int column;
+  std::string prefix;
+  std::string default_;
```

Comment:
Mostly personal opinion (I guess), but I _dislike_ publicly visible members that end in an underscore.
Thoughts on naming this one `default_value` instead?

Reply:
I have nothing against that. I just usually put an underscore at the end of keyword-names if nothing else comes to mind.